File: udemy/queue/queue.py
from typing import Any
# ライブラリも存在する(appendとpopleft)、popを使えばstackのような使い方もできる
# from collections import deque


class Queue(object):
    def __init__(self) -> None:
        self.queue = []

    def enqueue(self, data) -> None:
        self.queue.append(data)

    def dequeue(self) -> Any:
        if self.queue:
            return self.queue.pop(0)


if __name__ == '__main__':
    queue = Queue()
    queue.enqueue(1)
    queue.enqueue(2)
    queue.enqueue(3)
    queue.dequeue()
    print(queue.queue)

File: udemy/queue/queue.ts
class Queue {
  queue: number[];

  constructor() {
    this.queue = [];
  }

  enqueue(val: number) {
    this.queue.push(val);
  }

  dequeue() {
    if (this.queue) {
      this.queue.shift();
    }
  }
}

const quque = new Queue();
quque.enqueue(1);
quque.enqueue(2);
quque.enqueue(4);
quque.enqueue(5);
quque.dequeue();

console.log(quque.queue);
